miner/src/block_assembler.rs

```diff
@@ -200,21 +203,19 @@ impl BlockAssembler {
     }
 
     fn transform_uncle(uncle: UncleBlock) -> UncleTemplate {
-        let UncleBlock { header, proposals } = uncle;
-
         UncleTemplate {
-            hash: header.hash().to_owned(),
+            hash: uncle.calc_header_hash(),
```

Comment:
The old implementation header.hash().to_owned()` is a cached hash, now need to calculate the hash every time?